test/cql-pytest/test_tombstone_limit.py

```diff
@@ -306,8 +306,9 @@ def test_static_row_tombstone_prefix(cql, test_keyspace, lowered_tombstone_limit
         statement = SimpleStatement(f"SELECT * FROM {table}", fetch_size=10)
         check_pages_many_partitions(cql.execute(statement), {-1: all_pks[-1]})
 
-        statement = SimpleStatement(f"SELECT * FROM {table} WHERE v = 0 ALLOW FILTERING", fetch_size=10)
-        check_pages_many_partitions(cql.execute(statement), {-1: all_pks[-1]})
+        # The filtering counterpart is not possible to test after #20485 is fixed
```

Comment:
I don't understand... Is it now "not possible to test" or this test was buggy in the first place and should never have tested this incorrect case? If the latter is true, you should outright delete those lines (and perhaps try to explain better why).
CC @denesb, who wrote this test.

Reply:
I don't know how to formulate a similar filter expression without running into other bugs.

Reply:
@denesb please help out

Reply:
I don't understand why the below query is not legal anymore. `v` is a regular column and I want to filter the tables for the rows where `v` is a certain value. Is this illegal now?

Reply:
It's not illegal, it just returns no rows, failing the test.

The population phase doesn't set v at all, so the query shouldn't return any rows. But we had a bug that static rows didn't consider restrictions on regular columns, so the WHERE clause was completely ignored.

Reply:
Right, the solution is to simply set `v` to some value in the `insert` phase, so there is something to return.

Reply:
Well, why didn't I think of that.

Reply:
But then, will the test be valid? It's testing tombstones on static rows, and we will have regular rows.

Reply:
The test uses partition tombstones. So with or without `v`, the partition is dead and the replica should page on those dead partitions.

Reply:
But what about the first and last partitions? They're live, and they select static rows.